apps/studio/lib/navigation.ts

```diff
@@ -0,0 +1,66 @@
+import type { NextRouter } from 'next/router'
+import { BASE_PATH } from './constants'
+import { useRouter } from 'next/navigation'
+
+type Router = NextRouter | ReturnType<typeof useRouter>
+
+const MIDDLE_MOUSE_BUTTON = 1
+
+/**
+ * Creates a navigation handler that supports keyboard, modifier clicks, and middle mouse button.
+ *
+ * This is a curried function that takes a URL and router, and returns an event handler that:
+ * - Handles keyboard navigation (Enter/Space keys)
+ * - Opens in new tab on Cmd/Ctrl + click
+ * - Opens in new tab on middle mouse button click
+ * - Performs normal navigation on regular click
+ *
+ * @param url - The relative URL to navigate to (e.g., "/project/123/functions/my-function")
+ * @param router - Next.js router instance (supports both Pages Router and App Router)
+ * @returns Event handler function for onClick, onAuxClick, and onKeyDown
+ *
+ * @example
+ * ```tsx
+ * const router = useRouter()
+ * const handleNavigation = createNavigationHandler(`/project/${ref}/functions/${slug}`, router)
+ *
+ * <TableRow
+ *   onClick={handleNavigation}
+ *   onAuxClick={handleNavigation}
+ *   onKeyDown={handleNavigation}
+ *   tabIndex={0}
+ * />
+ * ```
+ */
+export const createNavigationHandler = (url: string, router: Router) => {
+  return (event: React.MouseEvent | React.KeyboardEvent) => {
+    // Handle keyboard events
+    if ('key' in event) {
+      if (event.key === 'Enter' || event.key === ' ') {
```

Comment:
I think expected behavior is that Cmd + Enter also opens in new tab (like Cmd + click)